File: WoofIOS/SwiftPackages/Presentation/Profile/Tests/PresentationProfileTests/Factory/DogsFeedFactory.swift
import Combine
import DataSourceFakes
import Repositories
import Swinject
import SwinjectAutoregistration
import TestUtils
import Utils

final class DogsFeedFactory {
    private let container: Container
    private let repository: DogsRepository
    private var cancellables = Set<AnyCancellable>()

    init(_ container: Container) {
        self.container = container
        repository = container~>
    }

    @discardableResult
    func withLoadedDogs(count: Int) -> Self {
        DogsDataSourceFactory(container).withDogs(count: count)
        repository.getDogs(offset: 0, limit: count).pss_sink { _ in }.store(in: &cancellables)
        TimeAdvancingFactory(container).tick()
        return self
    }
}

File: WoofAndroid/presentation/profile/src/test/kotlin/com/perrystreet/woof/presentation/profile/factory/DogsFeedFactory.kt
package com.perrystreet.woof.presentation.profile.factory

import com.perrystreet.woof.datasource.dogs.DogsDataSourceFactory
import com.perrystreet.woof.repositories.dogs.DogsRepository
import org.koin.test.KoinTest
import org.koin.test.inject

class DogsFeedFactory : KoinTest {
    private val repository: DogsRepository by inject()

    fun withLoadedDogs(count: Int) = apply {
        DogsDataSourceFactory().withDogs(count)
        repository.getDogs(offset = 0, limit = count).ignoreElement().blockingAwait()
    }
}
